src/common/http/platform/beast/http.cpp

```diff
@@ -255,6 +255,31 @@ size_t GetContentLength(const PARSER &parser) {
 	}
 }
 
+expected::ExpectedBool HasBody(
+	const expected::ExpectedString &content_length,
+	const expected::ExpectedString &transfer_encoding) {
+	if (transfer_encoding) {
+		if (transfer_encoding.value() != "chunked") {
+			return expected::unexpected(error::Error(
+				make_error_condition(errc::not_supported),
+				"Unsupported Transfer-Encoding: " + transfer_encoding.value()));
+		}
+		return true;
+	}
+
+	if (content_length) {
+		auto length = common::StringToLongLong(content_length.value());
+		if (!length || length.value() < 0) {
+			return expected::unexpected(error::Error(
+				length.error().code,
+				"Content-Length contains invalid number: " + content_length.value()));
+		}
+		return length.value() > 0;
+	}
```

Comment:
We can use here also `GetContentLength`, right?

Reply:
Not here, unfortunately. `GetContentLength` operates on `optional<size_t>`, whereas this operates on `ExpectedString`. The reason for this difference is that the former comes from data that Beast has interpreted, whereas the latter comes from raw header data.

Reply:
Thank you for clarifying.